### PR TITLE
Adapt UART library.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/io.toit [hunk 1]
```diff
@@ -951,6 +980,21 @@ abstract mixin OutMixin:
...
 
+  /**
+  Closes the writer if it exists.
+
+  The $out $Writer doesn't have a 'close' method. However, we can set
+    the internal boolean to `closed`, so that further writes throw an exception, or
+    that existing writes are aborted.
+
+  Any existing write needs to be aborted by the caller of this method.
+    The `try-write_` should either throw or return the number of bytes that have been
```
Comment (kasperl) [inline lib/io.toit:991]:
```suggestion
    The $try-write_ should either throw or return the number of bytes that have been
```

File: lib/io.toit [hunk 2]
```diff
@@ -951,6 +980,21 @@ abstract mixin OutMixin:
       out_ = result
     return result
 
+  /**
+  Closes the writer if it exists.
+
+  The $out $Writer doesn't have a 'close' method. However, we can set
+    the internal boolean to `closed`, so that further writes throw an exception, or
```
Comment (kasperl) [inline lib/io.toit:987]: Why is closed in backticks?

File: lib/io.toit [hunk 3]
```diff
@@ -1028,6 +1072,19 @@ abstract mixin InMixin:
       in_ = result
     return in_
 
+  /**
+  Closes the writer if it exists.
+
+  The $in $Reader doesn't have a 'close' method. However, we can set
+    the internal boolean to `closed`, so that further reads return 'null'.
```
Comment (kasperl) [inline lib/io.toit:1079]: Why is closed in backticks?

File: lib/uart.toit [hunk 4]
```diff
@@ -295,6 +315,64 @@ class HostPort extends Port:
   set-control-flags flags/int:
     uart-set-control-flags_ uart_ flags
 
+class UartWriter extends Object with io.Writer:
+  port_/Port
+
+  constructor.private_ .port_:
+
+  /**
+  Writes data to the Port.
```
Comment (kasperl) [inline lib/uart.toit:324]:
```suggestion
  Writes data to the $Port.
```

File: lib/uart.toit [hunk 5]
```diff
@@ -295,6 +315,64 @@ class HostPort extends Port:
   set-control-flags flags/int:
     uart-set-control-flags_ uart_ flags
 
+class UartWriter extends Object with io.Writer:
```
Comment (kasperl) [inline lib/uart.toit:318]: Does this need to be public?

File: lib/uart.toit [hunk 6]
```diff
@@ -125,13 +127,17 @@ class Port implements reader.Reader:
...
-     state_ = ResourceState_ group uart_
+      --baud-rate/int
+      --data-bits/int=8
+      --stop-bits/StopBits=STOP-BITS-1
+      --parity/int=PARITY-DISABLED:
+    group := resource-group_
+    uart_ = uart-create-path_ group device baud-rate data-bits stop-bits.value_ parity
+    state_ = ResourceState_ group uart_
+
+  out -> UartWriter:
```
Comment (kasperl) [inline lib/uart.toit:138]: Do we need to type this with UartWriter?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/io.toit (hunk 2 above)
Comment (floitsch) [reply lib/io.toit]: removed.

File: lib/io.toit (hunk 1 above)
Comment (floitsch) [reply lib/io.toit]: done.

File: lib/io.toit (hunk 3 above)
Comment (floitsch) [reply lib/io.toit]: done.

**Pull request merged**